synapse/lib/stormlib/aha.py

```diff
@@ -531,5 +656,226 @@ async def _methPoolSvcDel(self, svcname):
             }
         }
         '''
-    }
+    },
+    {
+        'name': 'aha.svc.mirror',
+        'descr': textwrap.dedent('''\
+            Query the AHA services and their mirror relationships.
+
+            Note: non-mirror services are not displayed.
+        '''),
+        'cmdargs': (
+            ('--timeout', {'help': 'The number of seconds to wait for the mirrors to sync.',
+                           'default': 60, 'action': 'store'}),
+        ),
+        'storm': '''
+        init {
+            $conf = ({
+                "columns": [
+                    {"name": "name", "width": 40},
+                    {"name": "role", "width": 9},
+                    {"name": "online", "width": 7},
+                    {"name": "ready", "width": 6},
+                    {"name": "host", "width": 16},
+                    {"name": "port", "width": 8},
+                    {"name": "version", "width": 12},
+                    {"name": "nexus idx", "width": 10},
+                ],
+                "separators": {
+                    "row:outline": false,
+                    "column:outline": false,
+                    "header:row": "#",
+                    "data:row": "",
+                    "column": "",
+                },
+            })
+            $printer = $lib.tabular.printer($conf)
+            $timeout = $cmdopts.timeout
+        }
+
+        function get_cell_infos(vname) {
+            $cell_infos = ({})
+            for $info in $lib.aha.callPeerApi($vname, getCellInfo) {
```

Comment:
Can we pass a timeout value in here? That way AHA will eventually return ( in the event of a failure to construct the second connection to the target service ).

Reply:
Sure thing, do we want to use the cmdopt.timeout value?

Reply:
Ended up going with cmdopt.timeout.